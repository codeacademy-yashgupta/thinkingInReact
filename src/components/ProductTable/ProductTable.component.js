import React from 'react';
import ProductCategoryRow from './../ProductRowCategory/ProductRowCategory.component';
import ProductRow from './../ProductRow/ProductRow.component';

class ProductTable extends React.Component {
  static defaultProps = {
    products: 
      [ {category: "Sporting Goods", price: "$49.99", stocked: true, name: "Football"},
        {category: "Sporting Goods", price: "$9.99", stocked: true, name: "Baseball"},
        {category: "Sporting Goods", price: "$29.99", stocked: false, name: "Basketball"},
        {category: "Electronics", price: "$99.99", stocked: true, name: "iPod Touch"},
        {category: "Electronics", price: "$399.99", stocked: false, name: "iPhone 5"},
        {category: "Electronics", price: "$199.99", stocked: true, name: "Nexus 7"} ],
  }
  render() {
    const rows = [];
    let lastCategory = null;
    
    this.props.products.forEach((product) => {
      if (product.category !== lastCategory) {
        rows.push(
          <ProductCategoryRow
            category={product.category}
            key={product.category} />
        );
      }
      rows.push(
        <ProductRow
          product={product}
          key={product.name} />
      );
      lastCategory = product.category;
    });

    return (
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Price</th>
          </tr>
        </thead>
        <tbody>{rows}</tbody>
      </table>
    );
  }
}


export default ProductTable;