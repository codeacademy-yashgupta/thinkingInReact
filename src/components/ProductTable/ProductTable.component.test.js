import React from 'react';
import ProductTable from './ProductTable.component';
import renderer from 'react-test-renderer';

describe('Product Table', () => {
  it('should render the product table', () => {
    const tree = renderer.create(<ProductTable/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})