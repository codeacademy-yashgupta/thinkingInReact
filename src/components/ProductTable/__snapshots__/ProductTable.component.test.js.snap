// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Table should render the product table 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th
        colSpan="2"
      >
        Sporting Goods
      </th>
    </tr>
    <tr>
      <td>
        <span
          style={
            Object {
              "color": "black",
            }
          }
        >
          Football
        </span>
      </td>
      <td>
        $49.99
      </td>
    </tr>
    <tr>
      <td>
        <span
          style={
            Object {
              "color": "black",
            }
          }
        >
          Baseball
        </span>
      </td>
      <td>
        $9.99
      </td>
    </tr>
    <tr>
      <td>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
          Basketball
        </span>
      </td>
      <td>
        $29.99
      </td>
    </tr>
    <tr>
      <th
        colSpan="2"
      >
        Electronics
      </th>
    </tr>
    <tr>
      <td>
        <span
          style={
            Object {
              "color": "black",
            }
          }
        >
          iPod Touch
        </span>
      </td>
      <td>
        $99.99
      </td>
    </tr>
    <tr>
      <td>
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
          iPhone 5
        </span>
      </td>
      <td>
        $399.99
      </td>
    </tr>
    <tr>
      <td>
        <span
          style={
            Object {
              "color": "black",
            }
          }
        >
          Nexus 7
        </span>
      </td>
      <td>
        $199.99
      </td>
    </tr>
  </tbody>
</table>
`;
